Lib/Search/ElasticSearch/Client.php

```diff
@@ -152,8 +153,8 @@ public function removeDocuments(ClassMetadata $class, array $documents)
         }
 
         foreach ($idsByIndex as $index => $ids) {
-            $type = $this->getIndex($index)->getType($class->type);
-            $type->deleteByQuery(new Query\Terms('id', $ids));
+            $query = new Query\Terms('id', $ids);
```

Comment:
`id` may not be the right identifier. You can get the identifier from the `ClassMetadata`.